providers/implementations/macs/kmac_prov.c

```diff
@@ -296,10 +307,12 @@ static int kmac_init(void *vmacctx, const unsigned char *key,
         (void)kmac_set_ctx_params(kctx, cparams);
     }
 
-    return bytepad(out, &out_len, kmac_string, sizeof(kmac_string),
-                   kctx->custom, kctx->custom_len, block_len)
-           && EVP_DigestUpdate(ctx, out, out_len)
-           && EVP_DigestUpdate(ctx, kctx->key, kctx->key_len);
+    res = bytepad(out, &out_len, kmac_string, sizeof(kmac_string),
```

Comment:
probably wouldnt hurt to assert that out_max >= out_len

Reply:
No need with the extra call to bytepad() to figure out the size.